### DISCUSSION TITLE
Installation freeze after: "Using VPN: WireGuard" or "Using VPN: OpenVPN"

### DISCUSSION THREAD

**Discussion by romton843** [pivpn/pivpn#1520] [Support] [answered]
Hi,
I'm trying to install PiVPN on Bullseye 64 bit and everything works perfectly up to the choice of VPN and no matter the choice I make between WireGuard and OpenVPN, it stops right there, nothing happens afterward. Is it a common problem that you know how to solve or should I open an issue? 
I already saw one comment about this issue but couldn't find any solution in the issues or discussions.
Romain
Here is the whole process:

pi@Romanopi:~ $ curl -L https://install.pivpn.io | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   162  100   162    0     0    295      0 --:--:-- --:--:-- --:--:--   295
100 92976  100 92976    0     0  93443      0 --:--:-- --:--:-- --:--:-- 93443
:::
::: sudo will be used for the install.
::: Hostname length OK
::: Verifying free disk space...
df: /run/user/1000/doc: Operation not permitted
:::
::: Package Cache update is needed, running apt-get update -y ...
 done!
:::
::: Checking apt-get for upgraded packages.... done!
:::
::: Your system is up to date! Continuing with PiVPN installation...
:::    Checking for git... already installed!
:::    Checking for tar... already installed!
:::    Checking for curl... already installed!
:::    Checking for grep... already installed!
:::    Checking for dnsutils... already installed!
:::    Checking for grepcidr... already installed!
:::    Checking for whiptail... already installed!
:::    Checking for net-tools... already installed!
:::    Checking for bsdmainutils... already installed!
:::    Checking for bash-completion... already installed!
:::    Checking for iptables-persistent... already installed!
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
::: IPv6 test connections to google.com have failed. Disabling IPv6 support. (The curl test failed with code: 28)
::: Using interface: eth0
::: Using User: pi
:::
::: Checking for existing base files...
:::    Checking /usr/local/src/pivpn is a repo... OK!
:::     Updating repo in /usr/local/src/pivpn from https://github.com/pivpn/pivpn.git ... done!
::: Using VPN: WireGuard

**Comment by romton843** [comment]
By the way, the template for support request doesn't exist. Should I fill the template for issues?

**Comment by romton843** [comment, marked as answer]
Nevermind, it works with sudo in front of curl -L https://install.pivpn.io/ | bash
Sorry for bothering and thank you all for your awesome work.
Romain

**Comment by ymralin** [comment]
I have the same problem and adding sudo changes nothing.
I installed pivpn once, but i uninstalled it as i wanted to change something. This issue is present when i try to install it again.

Unfortunatelly i have no backup of the system from before the installation and i'd rather not reinstall the OS, as the pi is running some other stuff.

**Comment by captaintinker** [comment]
I've also got this same issue.  I've tried it on a brand new Raspberry Pi 4b 8 gb, using the latest bullseye based Debian release from Raspberry Pi, both 32 and 64 bit, lite, and standard versions.  I've even used it on a PC based virtual machine using the buster Debian ISO from Raspberry Pi, and each time, regardless of the operating system, or how much I've updated it, or if I use sudo, or if I do it as root, the install stops at the same spot, right after choosing between Wireguard or OpenVPN.  It doesn't matter which one I try, it exits the installer and hangs until I hit ctrl-c.  I'm trying to decide if it's the script itself (curl -L https://install.pivpn.io | bash), or if it's something to do with the information I'm putting in it?  Those seem to be the only common things.  My IP address range on my internal network is a 10.x.x.x network, not the typical 192.168.x.x network.  I don't know if that has anything to do with it?

**Reply by ymralin** [reply]
The workaround from here: https://github.com/pivpn/pivpn/issues/1518 worked for me and i was able to install pivpn, but i'm unable to configure it to work properly.

**Reply by captaintinker** [reply]
Interesting...  I actually saw that before, but wasn't sure what to do about it.  It took me a while to figure out where the script was.  (/usr/local/src/pivpn/auto_install/install.sh)  I edited the script and looked for the generateRandomSubnet function.  In the post, it mentioned he replaced the generateRandomSubnet Function with sth like echo "10.0.0.1".  I saw there was an echo command with echo "${pivpnNET}".  I tried to edit it to show my IP address, and ran the install.sh script.  That didn't seem to fix it.  When I went back to edit the script again, I noticed it was put back to the original.  I ended up commenting out the entire function, line by line.  Not sure if that's the right way to do it, but since I'm not an expert on scripts, I figured an educated guess wouldn't hurt.  He did suggest manually filtering out this edge case.  So I did.  I re-ran the script, and I was able to get past the hangup, and finish the install.  I haven't gone any further yet, so I don't know for sure if it's worked or not yet.  I'll post later once I've had the chance to try it out.

**Reply by ymralin** [reply]
You have to comment out (or delete) the whole content of that function and add that last "echo" at the end.

**Reply by captaintinker** [reply]
Ok, I tried deleting the whole function except for the echo, and put in my IP address.  It did allow me to complete the script again, but looking at the output, it looks like there are some issues still (in **Bold**):

root@rpi:/usr/local/src/pivpn/auto_install# sudo ./install.sh
10.X.X.X
:::
::: You are root.
::: Hostname length OK
::: Verifying free disk space...
:::
::: Package Cache update is needed, running apt-get update -y ...
 done!
:::
::: Checking apt-get for upgraded packages.... done!
:::
::: There are 5 updates available for your system!
::: We recommend you update your OS after installing PiVPN!
:::
:::    Checking for git... already installed!
:::    Checking for tar... already installed!
:::    Checking for curl... already installed!
:::    Checking for grep... already installed!
:::    Checking for dnsutils... already installed!
:::    Checking for grepcidr... already installed!
:::    Checking for whiptail... already installed!
:::    Checking for net-tools... already installed!
:::    Checking for bsdmainutils... already installed!
:::    Checking for bash-completion... already installed!
:::    Checking for dhcpcd5... already installed!
:::    Checking for iptables-persistent... already installed!
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 5 not upgraded.
::: IPv6 test connections to google.com have failed. Disabling IPv6 support. (The curl test failed with code: 7)
::: Using User: pi
:::
::: Checking for existing base files...
:::    Checking /usr/local/src/pivpn is a repo... OK!
:::     Updating repo in /usr/local/src/pivpn from https://github.com/pivpn/pivpn.git ... done!
::: Using VPN: WireGuard
**./install.sh: line 1245: generateRandomSubnet: command not found**
::: Installing WireGuard from Debian package...
:::    Checking for wireguard-tools... already installed!
:::    Checking for qrencode... already installed!
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 5 not upgraded.
  [i] Listening on all interfaces, permitting origins from one hop away (LAN)
  [✓] Restarting DNS server
::: Backing up the wireguard folder to /etc/wireguard_2022-06-08-112408.tar.gz
::: Server Keys have been generated.
::: Server config generated.
**iptables v1.8.7 (nf_tables): host/network `' not found
Try `iptables -h' or 'iptables --help' for more information.**
::: Install Complete...
::: Restarting services...
**Job for wg-quick@wg0.service failed because the control process exited with error code.
See "systemctl status wg-quick@wg0.service" and "journalctl -xe" for details.**
:::    Checking for unattended-upgrades... not installed!
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  python3-dbus python3-distro-info
Suggested packages:
  python-dbus-doc python3-dbus-dbg bsd-mailx default-mta | mail-transport-agent needrestart powermgmt-base python3-gi
Recommended packages:
  python3-gi
The following NEW packages will be installed:
  python3-dbus python3-distro-info unattended-upgrades
0 upgraded, 3 newly installed, 0 to remove and 5 not upgraded.
Need to get 0 B/202 kB of archives.
After this operation, 798 kB of additional disk space will be used.
Preconfiguring packages ...
Selecting previously unselected package python3-dbus.
(Reading database ... 59113 files and directories currently installed.)
Preparing to unpack .../python3-dbus_1.2.16-5_arm64.deb ...
Unpacking python3-dbus (1.2.16-5) ...
Selecting previously unselected package python3-distro-info.
Preparing to unpack .../python3-distro-info_1.0_all.deb ...
Unpacking python3-distro-info (1.0) ...
Selecting previously unselected package unattended-upgrades.
Preparing to unpack .../unattended-upgrades_2.8_all.deb ...
Unpacking unattended-upgrades (2.8) ...
Setting up python3-dbus (1.2.16-5) ...
Setting up python3-distro-info (1.0) ...
Setting up unattended-upgrades (2.8) ...

Creating config file /etc/apt/apt.conf.d/20auto-upgrades with new version

Creating config file /etc/apt/apt.conf.d/50unattended-upgrades with new version
Created symlink /etc/systemd/system/multi-user.target.wants/unattended-upgrades.service -> /lib/systemd/system/unattended-upgrades.service.
Synchronizing state of unattended-upgrades.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable unattended-upgrades
Processing triggers for man-db (2.9.4-2) ...
:::    Package unattended-upgrades successfully installed!
::: Setupfiles copied to /etc/pivpn/wireguard/setupVars.conf
::: Installing scripts to /opt/pivpn...
 done.
::: Flushing writes to disk...
::: done.
:::
